include/drivers/st25r3911b_nfca.h

```diff
@@ -0,0 +1,321 @@
+	 *  @param err Transfer error. If 0, the transfer was successful.
+	 */
+	void (*transfer_completed)(const u8_t *data, size_t len, int err);
+
+	/** @brief NFC-A receive time-out.
+	 *
+	 *  @param tag_sleep If set, the tag is in sleep mode.
+	 */
+	void (*rx_timeout)(bool tag_sleep);
+};
+
+/** @brief Initialize NFC Reader NFC-A technology.
+ *
+ *  @details This function initializes everything that is required,
+ *           like interrupts and common reader functionality.
+ *
+ *  @param[in] events NFC-A Events.
+ *  @param[in] cnt Event count. This driver needs 2 events.
+ *  @param[in] cb Callback structure.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_init(struct k_poll_event *events, u8_t cnt,
+			  const struct nfca_cb *cb);
+
+/** @brief Switch on the NFC Reader field.
+ *
+ *  @details If the field is switched on successfully,
+ *           the @ref nfca_cb.field_on callback is
+ *           called.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_field_on(void);
+
+/** @brief Switch off the NFC Reader field.
+ *
+ *  @details If the field is switched off successfully,
+ *           the @ref nfca_cb.field_off callback is
+ *           called.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_field_off(void);
+
+/** @brief Detect tag by sending a detection command.
+ *
+ *  @details The command is sent according to NFC Forum Digital 2.0
+ *           6.6. After success, the @ref nfca_cb.tag_detected
+ *           callback is called.
+ *
+ *  @param cmd Command.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_tag_detect(enum nfca_detect_cmd cmd);
+
+/** @brief NFC-A collision resolution procedure.
+ *
+ *  @details The collision resolution procedure is described in
+ *           NFC Forum Activity 2.0 9.4.4. After this
+ *           procedure, @ref nfca_cb.anticollision_completed
+ *           is called.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_anticollision_start(void);
+
+/** @brief Exchange the specified amount of data.
+ *
+ *  @details This is a generic function for NFC-A data exchange.
+ *
+ *  @param[in] tx TX data buffer.
+ *  @param[in,out] rx RX data buffer.
+ *  @param[in] auto_crc If set, the CRC is generated automatically by the
+ *                      NFC Reader. Otherwise, the CRC must be added
+ *                      manually to the TX buffer.
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int st25r3911b_nfca_transfer(struct nfca_buf *tx, struct nfca_buf *rx,
```

Comment:
Why aren't these pointers pointing to constant structures? Are the fields of these structures supposed to be modified in this function? I don't see them modified in the code, but I might have missed something.

Reply:
Fixed.